### PR TITLE
[14.0][IMP] compute product's outgoing and incoming qty by restrict_lot_id

### PR DESCRIPTION
**Pull request by mt-software-de**
To be able to get the outgoing qty and the incoming qty.
For a specific lot i changed the _get_domain_locations
This was the easiest way to change the move_in_domain and move_out_domain
which is used in _compute_quantities_dict https://github.com/odoo/odoo/blob/14.0/addons/stock/models/product.py#L123

### CODE REVIEW

**Comment by mt-software-de** [comment]
@jbaudoux

File: stock_restrict_lot/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    # To be able to get the outgoing and incoming qty by the restrict_lot_id
+    # the odoo standard method _compute_quantities_dict is hookable
+    # therefore the domain is already here changed
+    def _get_domain_locations(self):
+        res = super()._get_domain_locations()
+        restrict_lot_id = self.env.context.get("restrict_lot_id")
```
Comment (jbaudoux) [inline stock_restrict_lot/models/product_product.py]: It is lot_id in the context according to
https://github.com/odoo/odoo/blob/14.0/addons/stock/models/product.py#L121

```suggestion
        restrict_lot_id = self.env.context.get("lot_id")
```

File: stock_restrict_lot/tests/test_restrict_lot.py [hunk 2]
```diff
@@ -239,3 +239,35 @@ def assert_move_line_per_lot_and_location(
...
+
+        lot2 = self.lot.copy({"name": "lot2"})
+        move2 = move.copy()
+        move2.restrict_lot_id = lot2.id
+        move2._action_confirm()
+
+        product = move.product_id
+        self.assertEqual(product.outgoing_qty, 2)
+        product.invalidate_cache()
+        product = product.with_context(restrict_lot_id=self.lot.id)
```
Comment (jbaudoux) [inline stock_restrict_lot/tests/test_restrict_lot.py]:
```suggestion
        product = product.with_context(lot_id=self.lot.id)
```

File: stock_restrict_lot/tests/test_restrict_lot.py [hunk 3]
```diff
@@ -239,3 +239,35 @@ def assert_move_line_per_lot_and_location(
...
+        self.assertEqual(product.outgoing_qty, 1)
+
+        product.invalidate_cache()
+        product = product.with_context(restrict_lot_id=lot2.id)
```
Comment (jbaudoux) [inline stock_restrict_lot/tests/test_restrict_lot.py]:
```suggestion
        product = product.with_context(lot_id=lot2.id)
```

**Review by jbaudoux — CHANGES_REQUESTED** [review]

**Review by jbaudoux — APPROVED** [review]
LGTM, thanks

**Comment by jbaudoux** [comment]
@sebalix Can you review this one?

**Review by sebalix — APPROVED** [review]

**Comment by sebalix** [comment]
@florian-dacosta are you OK with this change?

**Comment by mt-software-de** [comment]
> @florian-dacosta are you OK with this change?

Any news on this?

**Comment by rousseldenis** [comment]
@florian-dacosta

**Comment by mt-software-de** [comment]
@sebalix @rousseldenis @jbaudoux how should we continue here?

**Review by florian-dacosta — COMMENTED** [review]
Hi @mt-software-de 
Sorry, it seems I missed the ping...multipe times!
The changes are ok for me, but it seems a bit too much to put yourself as author of the module IMO.

**Comment by mt-software-de** [comment]
> Hi @mt-software-de 
> Sorry, it seems I missed the ping...multipe times!
> The changes are ok for me, but it seems a bit too much to put yourself as author of the module IMO.

Yes for sure that is to much. I will change this.

**Comment by mt-software-de** [comment]
Could we now merge this one?

**Review by florian-dacosta — APPROVED** [review]

**Pull request merged**